src/idl/src/tree.c

```diff
@@ -772,6 +772,59 @@ uint32_t idl_bound(const void *node)
   return 0u;
 }
 
+bool idl_is_optional(const void *node)
+{
+  if (idl_is_declarator(node)) {
+    const idl_node_t *parent = idl_parent(node);
+    if (idl_is_member(parent))
+      return idl_is_optional(parent);
+  } else {
+    idl_annotation_appl_t *annotation_appl = NULL;
+    IDL_FOREACH(annotation_appl, ((const idl_node_t*)node)->annotations) {
+      if (0 != strcmp(idl_identifier(annotation_appl->annotation), "optional"))
```

Comment:
Although we don't have a coding style guide and slightly different styles are used through the code base, I'd prefer a consistent coding style in a single code file, so `strmp(..) != 0` or just `strcmp(...)`